packages/eas-cli/src/credentials/ios/api/GraphqlClient.ts

```diff
@@ -0,0 +1,233 @@
+      );
+    } else {
+      await IosAppBuildCredentialsMutation.setDistributionCertificateAsync(
+        iosAppBuildCredentials.id,
+        appleDistributionCertificateId
+      );
+      return await IosAppBuildCredentialsMutation.setProvisioningProfileAsync(
+        iosAppBuildCredentials.id,
+        appleProvisioningProfileId
+      );
+    }
+  }
+
+  public async createOrGetExistingAppleTeamAsync(
+    { account }: AppLookupParams,
+    { appleTeamIdentifier, appleTeamName }: { appleTeamIdentifier: string; appleTeamName: string }
+  ): Promise<AppleTeam> {
+    const appleTeam = await AppleTeamQuery.byAppleTeamIdentifierAsync(
+      account.id,
+      appleTeamIdentifier
+    );
+    if (appleTeam) {
+      return appleTeam;
+    } else {
+      return await AppleTeamMutation.createAppleTeamAsync(
+        { appleTeamIdentifier, appleTeamName },
+        account.id
+      );
+    }
+  }
+
+  public async createOrGetExistingAppleAppIdentifierAsync(
+    { account, bundleIdentifier }: AppLookupParams,
+    appleTeam: AppleTeam
+  ): Promise<AppleAppIdentifier> {
+    const appleAppIdentifier = await AppleAppIdentifierQuery.byBundleIdentifierAsync(
+      account.name,
+      bundleIdentifier
+    );
+    if (appleAppIdentifier) {
+      return appleAppIdentifier;
+    } else {
+      return await AppleAppIdentifierMutation.createAppleAppIdentifierAsync(
+        { bundleIdentifier, appleTeamId: appleTeam.id },
+        account.id
+      );
+    }
+  }
+
+  public async getDevicesForAppleTeamAsync(
+    { account }: AppLookupParams,
+    { appleTeamIdentifier }: AppleTeam
+  ): Promise<AppleDevice[]> {
+    return await AppleDeviceQuery.getAllByAppleTeamIdentifierAsync(account.id, appleTeamIdentifier);
+  }
+
+  public async createProvisioningProfileAsync(
+    { account }: AppLookupParams,
+    appleAppIdentifier: AppleAppIdentifier,
+    appleProvisioningProfileInput: {
+      appleProvisioningProfile: string;
+      developerPortalIdentifier?: string;
+    }
+  ): Promise<AppleProvisioningProfile> {
+    return await AppleProvisioningProfileMutation.createAppleProvisioningProfileAsync(
+      appleProvisioningProfileInput,
+      account.id,
+      appleAppIdentifier.id
+    );
+  }
+
+  public async getProvisioningProfileAsync(
+    appLookupParams: AppLookupParams,
+    appleTeam: AppleTeam,
+    iosDistributionType: IosDistributionType
+  ): Promise<AppleProvisioningProfile | null> {
+    const projectFullName = formatProjectFullName(appLookupParams);
+    const appleAppIdentifier = await this.createOrGetExistingAppleAppIdentifierAsync(
+      appLookupParams,
+      appleTeam
+    );
+    return await AppleProvisioningProfileQuery.getForAppAsync(projectFullName, {
+      appleAppIdentifierId: appleAppIdentifier?.id,
+      iosDistributionType,
+    });
+  }
+
+  public async updateProvisioningProfileAsync(
+    appleProvisioningProfileId: string,
+    appleProvisioningProfileInput: {
+      appleProvisioningProfile: string;
```

Comment:
should this also optionally take an `appleDeviceId` argument for adhoc provisioning profiles?

Reply:
I think not. 
https://github.com/expo/universe/blob/master/server/www/src/graphql/mutations/credentials/AppleProvisioningProfileMutation.ts#L12 This mutation doesn't expect apple device ids to be passed in.
The device IDs are resolved from the provisioning profile either in `AppleProvisioningProfileEntity.updateAsync` or `AppleProvisioningProfileEntity.createAsync` (https://github.com/expo/universe/blob/master/server/www/src/data/entities/credentials/AppleProvisioningProfileEntity.ts#L121)